tablib/packages/ods.py

```diff
@@ -0,0 +1,150 @@
+ <manifest:file-entry manifest:full-path="mimetype" manifest:media-type="text/plain"/>
+</manifest:manifest>''')
+        z.writestr('styles.xml', '''<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<office:document-styles xmlns:office="urn:oasis:names:tc:opendocument:xmlns:office:1.0">
+</office:document-styles>''')
+        self.content = tempfile.NamedTemporaryFile()
+        xml = self.xmlwriter = loxun.XmlWriter(self.content, pretty=False)
+        xml.addNamespace('office', OFFICE_NS)
+        xml.addNamespace('style', STYLE_NS)
+        xml.addNamespace('table', TABLE_NS)
+        xml.addNamespace('xlink', XLINK_NS)
+        xml.addNamespace('text', TEXT_NS)
+        # add bold style for headers
+        xml.startTag('office:document-content')
+        xml.startTag('office:automatic-styles')
+        xml.startTag('style:style', {
+            'style:family': 'paragraph',
+            'style:name': 'bold',
+            'style:display-name': 'bold',
+        })
+        xml.tag('style:text-properties', {
+            'style:font-weight-complex': 'bold',
+            'style:font-weight': 'bold',
+            'style:font-weight-asian': 'bold'
+        })
+        xml.endTag()
+        xml.endTag()
+        xml.startTag('office:body')
+        xml.startTag('office:spreadsheet')
+        self.status = self.OPENED
+
+    def close(self):
+        assert self.status == self.OPENED
+        self.status = self.CLOSED
+        xml = self.xmlwriter
+        xml.endTag()
+        xml.endTag()
+        xml.endTag()
+        self.z.write(self.content.name, 'content.xml')
+        self.content.close()
+        self.z.close()
+        del self.z
+        del self.xmlwriter
+        del self.content
+
+    def start_sheet(self, columns, title=None):
+        assert self.status == self.OPENED
+        self.status = self.INSHEET
+        xml = self.xmlwriter
+        attribs = {}
+        if title:
+            attribs['table:name'] = title
+        xml.startTag('table:table', attribs)
+        for i in range(columns):
+            xml.tag('table:table-column')
+
+    def end_sheet(self):
+        assert self.status == self.INSHEET
+        self.status = self.OPENED
+        self.xmlwriter.endTag()
+
+    def add_headers(self, headers):
+        self.add_row(headers, {
+            'table:style-name': 'bold',
+            'table:default-cell-style-name': 'bold',
+        }, hint='header')
+
+    def add_row(self, row, attribs={}, hint=None):
+        self.start_row(attribs)
+        for cell in row:
+            self.add_cell(cell, hint=hint)
+        self.end_row()
+
+    def start_row(self, attribs={}):
+        assert self.status == self.INSHEET
+        self.status = self.INROW
+        self.xmlwriter.startTag('table:table-row', attribs)
+
+    def end_row(self):
+        assert self.status == self.INROW
+        self.status = self.INSHEET
+        self.xmlwriter.endTag()
+
+    def add_cell(self, content, hint=None):
```

Comment:
Just an observation here. It is not a bug or anything.

`add_cell` does not support other cell data types, such as: int, float but unicode string.

Reply:
Yep, small amerliorations are still possible, I would do it if i had information from the maintainer that a possible integration is possible soon.

Reply:
Sent an invitation to you.